src/bundle/Controller/VersionController.php

```diff
@@ -35,22 +35,19 @@ public function __construct(
 
     public function contentViewTabAction(
         ContentView $view,
-        VersionFilter $versionFilter,
+        VersionParameterSupplier $versionParameterSupplier,
         UiFormFactory $formFactory
     ) {
-        $contentInfo = $view->getContent()->getVersionInfo()->getContentInfo();
-        $versions = $this->uiVersionService->loadVersions($contentInfo);
+        $versionParameterSupplier->supply($view);
 
-        $draftVersions = $versionFilter->filterDrafts($versions);
-        $draftActionsForm = $formFactory->createVersionsDraftActionForm($draftVersions);
-
-        $archivedVersions = $versionFilter->filterArchived($versions);
-        $archivedActionsForm = $formFactory->createVersionsArchivedActionForm($archivedVersions);
+        $draftActionsForm = $formFactory->createVersionsDraftActionForm(
+            $view->hasParameter('draftVersions') ? $view->getParameter('draftVersions') : []
```

Comment:
I really don't like that way of reusing what the supplier will hopefully set. There is no contract with the interface that certifies any of that.